pandas/core/generic.py

```diff
@@ -6602,17 +6603,20 @@ def replace(
 
                 # dest iterable dict-like
                 if is_dict_like(value):  # NA -> {'A' : 0, 'B' : -1}
-                    new_data = self._mgr
-
-                    for k, v in value.items():
-                        if k in self:
-                            new_data = new_data.replace(
-                                to_replace=to_replace,
-                                value=v,
-                                filter=[k],
-                                inplace=inplace,
-                                regex=regex,
-                            )
+
+                    # Operate column-wise
+                    res = self if inplace else self.copy()
+                    ax = self._info_axis
+                    for i in range(len(ax)):
+                        if ax[i] in value:
+                            v = value[ax[i]]
+                            ser = self._ixs(i, axis=1)
```

Comment:
same,

can you factor this to a function instead.

Reply:
refactored + green

i think in follow-up might be able to use this new helper method in one or two more places